src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +121,29 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        internal static TAttribute GetResolvedAttribute<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            var attribute = parameter.GetCustomAttribute<TAttribute>();
+
+            var attributeConnectionProvider = attribute as IConnectionProvider;
```

Comment:
IConnectionProvider allows us to generically interface with all the attributes to get/set their connection properties.